File: leetcode/0011.container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        res = 0
        i = 0
        j = len(height) - 1
        while ( i < j ):
            minHeight = min(height[i], height[j])
            dist = j - i
            area = dist * minHeight
            if area > res:
                res = area
            if height[i] < height[j]:
                i += 1
            else:
                j -= 1
        return res
        

File: leetcode/0011.container-with-most-water.ts
function maxArea(height: number[]): number {
    let result = 0
    let l = 0
    let r = height.length - 1
    while(l<r) {
        let length = r -l
        let breadth = Math.min(height[l], height[r])
        let area = length * breadth
        result = Math.max(result, area)
        if(height[l] < height[r]) {
            l += 1
        }
        else {
            r -= 1
        }
    }
    return result
};